src/parse/decl.rs

```diff
@@ -1795,6 +1800,14 @@ mod tests {
                 ArrayType::Unbounded
             )
         ));
+        assert!(match_type(
+            parse("int a[5];"),
+            Array(Box::new(Int(true)), ArrayType::Fixed(5))
+        ));
+        assert!(match_type(
+            parse("char b[static 42];"),
+            Array(Box::new(Char(true)), ArrayType::Fixed(42))
```

Comment:
Hmm, I just realized this is not quite correct ... `static` should only be valid inside of a function prototype, like this: `int f(int a[static 5]);`. Anywhere else it should give a `semantic_err`.

The way you can tell you are in a function is when `allow_abstract` is set. This was originally intended for knowing whether `int f(int)` should be allowed, but can be repurposed here.

`allow_abstract` is not currently passed into `postfix_type`, you will have to pass it from `direct_declarator`.

Reply:
The error clang gives is `'static' used in array declarator outside of function prototype` which I think is fine to reuse.